components/places/src/db/schema.rs

```diff
@@ -64,6 +64,13 @@ const CREATE_TABLE_HISTORYVISITS_SQL: &str =
         FOREIGN KEY(from_visit) REFERENCES moz_historyvisits(id)
     )";
 
+const CREATE_TABLE_HISTORYVISIT_TOMBSTONES_SQL: &str = "CREATE TABLE moz_historyvisit_tombstones (
```

Comment:
This is part of the privacy concern that [Richard brought up earlier](https://bugzilla.mozilla.org/show_bug.cgi?id=578694#c28). If we cleared all visits for the page, we'll forget it completely; but, as long as that Place is in your history, we'll not only keep those old visits around, but _also_ keep a record of the fact that you cleared them, and when.

Since there isn't a good way around that in existing Sync, though, I think: 1) resurrecting visits is definitely the more surprising behavior, 2) clearing history when it might have already been synced to other devices is already fraught, 3) private browsing mode exists, and 4) we _might_ be able to communicate this in the UI when you clear history.

I don't really have a suggestion here beyond "this vaguely concerns me, but the alternative concerns me more," though. 😉

What were your thoughts working on this?

Reply:
I feel like the tuple of (PlaceID, VisitTimestamp) has relatively few privacy issues TBH. It's also worth noting that this isn't synced, and disappears when the place is deleted.

Note that this isn't a guid, it's the local place ID.